src/bundle/EventListener/RequestAttributesListener.php

```diff
@@ -43,20 +43,21 @@ public static function getSubscribedEvents()
     /**
      * Adds all the request attributes to the parameters.
      *
-     * @param \eZ\Publish\Core\MVC\Symfony\View\Event\FilterViewBuilderParametersEvent $e
+     * @param FilterViewBuilderParametersEvent $event
      */
-    public function addRequestAttributes(FilterViewBuilderParametersEvent $e)
+    public function addRequestAttributes(FilterViewBuilderParametersEvent $event)
     {
-        if (!$this->isAdmin($e->getRequest())) {
+        $request = $event->getRequest();
+
+        if (!$this->isAdmin($request)) {
             return;
         }
 
-        $parameterBag = $e->getParameters();
+        $parameterBag = $event->getParameters();
 
-        if ($parameterBag->has('locationId')) {
+        if ($parameterBag->has('locationId') && $request->get('_route') === '_ezpublishLocation') {
```

Comment:
IMHO `$request->get('_route') === '_ezpublishLocation'`  is precondition to manipulate `$parameterBag` so if you move it into the first "if" statement, the code will be more readable 😉

Reply:
Yes, we discussed this with @webhdx. The only reason for this is that first check is lightweight